proto/buf/compiler/v1alpha1/ast.proto

```diff
@@ -0,0 +1,356 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.compiler.v1alpha1;
+
+import "buf/compiler/v1alpha1/report.proto";
+
+// NOTE: In this file, we number all Span-typed fields together, starting
+// at tag 10. They must also be suffixed with `_span`.
+
+// A parsed AST file. This is the root file for the whole Protocompile AST.
```

Comment:
My assumption was approximately "you should never ship around an AST because people will be naughty". I think kicking it into an internal package is probably sufficient.